model_analyzer/result/constraint_manager.py

```diff
@@ -100,31 +101,30 @@ def constraint_failure_percentage(
                     if ConstraintManager._metric_matches_constraint(
                             metric, constraints[i]):
                         failure_percentage += ConstraintManager._get_failure_percentage(
-                            metric, constraints[i][metric.tag])
+                            metric, constraints[i])
 
         return failure_percentage * 100
 
     @staticmethod
     def _metric_matches_constraint(
-            metric: Record, constraint: Dict[str, Dict[str, int]]) -> bool:
-        if constraint is not None and metric.tag in constraint:
+            metric: Record, constraint: ModelConstraints) -> bool:
+        if constraint is not None and constraint.has_metric(metric.tag):
```

Comment:
I don't think we need the `constraint is not None` code anymore

Reply:
Done suggested change